source/MaterialXTest/GenReference.cpp

```diff
@@ -51,7 +51,102 @@ const mx::ShaderPort* getShaderPort(const mx::ShaderStage& stage, const std::str
     return nullptr;
 }
 
-TEST_CASE("GenShader: OSL Reference", "[genshader]")
+TEST_CASE("GenReference: Reference implementation file test", "[genreference]")
```

Comment:
@niklasharrysson , this is WIP to put out the implementation glue file (nodedef->impls) for easier migration. Or at least to compare. It includes sanity checks on nodedefs.